app/routes/case.$caseId.tsx

```diff
@@ -0,0 +1,40 @@
+import { asImageSrc, asText, asLink } from "@prismicio/client";
+import { LoaderFunctionArgs } from "@remix-run/node";
+import { MetaFunction, useLoaderData } from "@remix-run/react";
+import { ProjectDocument } from "types.generated";
+import Case from "~/components/projects/case";
+import { getPrismicClient } from "~/utils/prismicio";
+
+export const meta: MetaFunction = () => {
+  return [
+    { title: "Case Study" },
+    { name: "description", content: "Welcome to Remix!" },
+  ];
+};
+
+export const loader = async ({ params }: LoaderFunctionArgs) => {
+  const caseId = params.caseId;
+  const client = await getPrismicClient();
+  const document = await client.getSingle<ProjectDocument>("project");
+  return { document, caseId };
+};
+
+export default function Projects() {
+  const { document, caseId } = useLoaderData<{
+    document: ProjectDocument;
+    caseId: string;
+  }>();
+  const project = document.data.project.find(
+    (project) => asText(project.title) === caseId,
+  );
+  const caseDetails = {
+    logo: asImageSrc(project?.logo_image) ?? undefined,
+    title: asText(project?.title),
+    introduction: asText(project?.description),
+    problem_statement: asText(project?.case_problem),
+    solution: asText(project?.case_problem),
```

Comment:
it looks like you're using the ?? to default to undefined for some of the props, should we just default to it for all of them and keep the original case types?